### PR TITLE
Migrate SO3 to use `torchlie`

### PR DESCRIPTION
**Pull request by luisenp**

### CODE REVIEW

**Review by fantaosha — APPROVED** [review]
LGTM.

File: theseus/geometry/se3.py [hunk 1]
```diff
@@ -583,7 +583,7 @@ def transform_from(
             # Right jacobians for SE3 are computed
             Jg = torch.zeros(batch_size, 3, 6, dtype=self.dtype, device=self.device)
             Jg[:, :, :3] = self[:, :, :3]
-            Jg[:, :, 3:] = -self[:, :, :3] @ SO3.hat(p)
+            Jg[:, :, 3:] = -self[:, :, :3] @ SO3.hat(p.squeeze(-1))
```
[the remark this answers is not in the document]
Comment (luisenp) [reply theseus/geometry/se3.py:586]: Shape check is the same as we had before. Looks like this allows arbitrary size for `dim=2`?

image

File: theseus/geometry/so3.py [hunk 2]
```diff
@@ -46,34 +46,16 @@ def rand(
...
-                a * torch.sin(u2),
-                a * torch.cos(u2),
-                b * torch.sin(u3),
-                b * torch.cos(u3),
-            ],
-            dim=1,
-        )
-        assert quaternion.shape == (size[0], 4)
-        return SO3(quaternion=quaternion)
+        return SO3(tensor=tensor)
```
[the remark this answers is not in the document]
Comment (luisenp) [reply theseus/geometry/so3.py:58]: I don't think so. We could since this is coming from `torchlie`, right? But seems like using the context is the only way?

File: theseus/geometry/se3.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fantaosha) [reply theseus/geometry/se3.py:586]: I forgot why we made this change. If `p.shape=[..., 3, 1]`, this operator does not make sense.

File: theseus/geometry/so3.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fantaosha) [reply theseus/geometry/so3.py:58]: That's correct. Does the constructor have an option for this? We seemed to have one before but it was removed.

**Review by mhmukadam — COMMENTED** [review]
LGTM
